conda-store-server/tests/test_db_api.py

```diff
@@ -39,35 +39,104 @@ def test_namespace_crud(db):
 
 def test_namespace_role_mapping(db):
     namespace_name = "pytest-namespace"
+    other_namespace_name1 = "pytest-other-namespace1"
+    other_namespace_name2 = "pytest-other-namespace2"
+    other_namespace_name3 = "pytest-other-namespace3"
 
-    # starts with no namespaces for test
+    # Starts with no namespaces
     assert len(api.list_namespaces(db).all()) == 0
 
-    # create namespace
+    # Creates namespaces
     namespace = api.create_namespace(db, name=namespace_name)
+    other_namespace1 = api.create_namespace(db, name=other_namespace_name1)
+    other_namespace2 = api.create_namespace(db, name=other_namespace_name2)
+    other_namespace3 = api.create_namespace(db, name=other_namespace_name3)
     db.commit()
 
-    # check that only one namespace exists
-    assert len(api.list_namespaces(db).all()) == 1
+    # Checks that all namespaces exist
+    assert len(api.list_namespaces(db).all()) == 4
+
+    # Creates role mappings
+    api.create_namespace_role(db, name=namespace_name, other=other_namespace_name1, role="admin")
+    api.create_namespace_role(db, name=namespace_name, other=other_namespace_name2, role="admin")
+    api.create_namespace_role(db, name=namespace_name, other=other_namespace_name3, role="viewer")
+    db.commit()
```

Comment:
admin is repeated twice on purpose - I overwrite it in this test later. This also makes sure we don't somehow not allow duplicate roles without considering other keys.